components/dash-core-components/tests/integration/markdown/test_markdown.py

```diff
@@ -90,6 +91,278 @@ def test_mkdw002_dcclink(dash_dcc):
+    assert not dash_dcc.driver.execute_script("return !!window.MathJax")
+    assert dash_dcc.get_logs() == []
+
+
+@pytest.mark.parametrize("is_eager", [True, False])
+def test_mkdw004_inline_mathjax(dash_dcc, is_eager):
+    app = Dash(__name__, eager_loading=is_eager, assets_folder="../../assets")
+
+    app.layout = html.Div(
+        [
+            dcc.Markdown("# h1 tag with inline MathJax: $E=mc^2$", mathjax=True),
+        ]
+    )
+
+    dash_dcc.start_server(app)
+    assert dash_dcc.get_logs() == []
+
+
+@pytest.mark.parametrize("is_eager", [True, False])
+def test_mkdw005_block_mathjax(dash_dcc, is_eager):
+    app = Dash(__name__, eager_loading=is_eager, assets_folder="../../assets")
+
+    app.layout = html.Div(
+        [
+            dcc.Markdown(
+                """
+                    ## h2 tag with MathJax block:
+                    $$
+                    \\frac{1}{(\\sqrt{\\phi \\sqrt{5}}-\\phi) e^{\\frac25 \\pi}} =
+                    1+\\frac{e^{-2\\pi}} {1+\\frac{e^{-4\\pi}} {1+\\frac{e^{-6\\pi}}
+                    {1+\\frac{e^{-8\\pi}} {1+\\ldots} } } }
+                    $$
+                    ## Next line.
+                """,
+                mathjax=True,
+            ),
+        ]
+    )
+
+    dash_dcc.start_server(app)
+    assert dash_dcc.get_logs() == []
+
+
+@pytest.mark.parametrize("is_eager", [True, False])
+def test_mkdw006_toggle_mathjax(dash_dcc, is_eager):
+    app = Dash(__name__)
+
+    gravity = "$F=\\frac{Gm_1m_2}{r^2}$"
+
+    app.layout = html.Div(
+        [
+            html.Button("Toggle MathJax", id="btn"),
+            dcc.Markdown(
+                f"""
+                    # Test MathJax Toggling {gravity}
+                """,
+                id="md",
+            ),
+        ]
+    )
+
+    @app.callback(
+        Output("md", "mathjax"), Input("btn", "n_clicks"), prevent_initial_call=True
+    )
+    def toggle(n):
+        return (n or 0) % 2 != 0
+
+    dash_dcc.start_server(app)
+
+    # Initial state: no MathJax loaded or rendered, unformatted text is shown
+    dash_dcc.wait_for_contains_text("#md", gravity)
+    dash_dcc.wait_for_no_elements("#md svg")
+    assert not dash_dcc.driver.execute_script("return !!window.MathJax")
+
+    btn = dash_dcc.find_element("#btn")
+    btn.click()
+
+    # One click: MathJax is rendered, unformatted text is gone
+
+    dash_dcc.wait_for_element("#md svg")
+    assert gravity not in dash_dcc._get_element("#md").text
+    assert dash_dcc.driver.execute_script("return !!window.MathJax")
+
+    btn.click()
+
+    # Second click: Back to initial state except that MathJax library is still loaded
+    dash_dcc.wait_for_contains_text("#md", gravity)
+    dash_dcc.wait_for_no_elements("#md svg")
+    assert dash_dcc.driver.execute_script("return !!window.MathJax")
+
+
+@pytest.mark.parametrize("is_eager", [True, False])
+def test_mkdw007_load_mathjax(dash_dcc, is_eager):
+    app = Dash(__name__)
```

Comment:
```suggestion
    app = Dash(__name__, eager_loading=is_eager)
```
And again this will change the initial state of `!!window.MathJax`